### DISCUSSION TITLE
Feature Request: Support for $$NOW

### DISCUSSION THREAD

**Discussion by rubenvereecken** [typegoose/mongodb-memory-server#522] [Q&A] [answered]
## Versions

- mongodb-memory-server-*: 7.3.4

package: mongo-memory-server

## What is your question?

I found that there is no support for the mongodb aggregation variable [`$$NOW`](https://docs.mongodb.com/manual/reference/aggregation-variables/#mongodb-variable-variable.NOW) which simply returns the current time. We use it in a scoring function used in document sorting. Would it be possible to request this functionality?

**Comment by hasezoey** [comment, marked as answer]
like many others already asked, this package only manages the mongodb binary / instance

from what i can tell, variable `$$NOW` got added in 4.2, where this package's default is `4.0.25`, you will have to set your version to `4.2` with [`VERSION`](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options#version) ([read here on how to add it to the package.json](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options#how-to-use-them-in-the-packagejson)) or use the package `mongodb-memory-server-global-4.2`

**Reply by rubenvereecken** [reply]
Brilliant as always, thanks @hasezoey! I hope the issue can at least help some Google search results find their answers.

**Reply by rubenvereecken** [reply]
@hasezoey I'd like to point out that the following config didn't work
```json
  "mongodbMemoryServer": {
    "version": "4.4.7"
  },
```

But the env variable `MONGOMS_VERSION=4.4.7` did work. Hope it helps!

EDIT: Never mind I'll add the `config` later to the JSON

**Reply by hasezoey** [reply]
> I'd like to point out that the following config didn't work

~~did you wrap it in a `"config": {}` like mentioned in the documentation? when yes, what is you project / folder structure (MMS searched from CWD upwrads for a package.json that includes `config.mongodbMemoryServer`)?~~

nevermind, didnt see the EDIT
